src/common.jl

```diff
@@ -122,7 +122,7 @@ end
 # Weighted linear least squares
 _wlstsq(vand, y, W::Number) = _wlstsq(vand, y, fill!(similar(y), W))
 _wlstsq(vand, y, W::AbstractVector) = _wlstsq(vand, y, Diagonal(W))
-_wlstsq(vand, y, W::AbstractMatrix) = (vand' * W * vand) \ (vand' * W * y)
+_wlstsq(vand, y, W::AbstractMatrix) = qr(vand' * W * vand) \ (vand' * W * y)
```

Comment:
The handling of the weights `W` here puzzles me. I would expect the weights to mean that the polynomial is designed such that the norm of `W*(P.(x) - y)` is minimized, i.e. the norm of `W*(vand*c - y)` is minimized to yield the coefficients `c`. That could be solved using the normal equations `vand'*W'*W*vand*c = vand'*W'*W*y` (which is similar to the old code, but not quite it) or more directly using the QR decomposition as `c=qr(W*vand)\(W*y)`. Here, it looks like the weighting uses effectively the square roots of the weights in the interpretation I would have expected. Is that intentional?

Reply:
> Is that intentional?

Not on my part. I'm pretty sure I inherited that code or have certainly forgot what I intended, if I added it. I'll have to look into it and find a reference. Thanks for bringing it up.

Reply:
Yes, this is specifying "W" as W'W which basically means the weights are specified as squares. This is different from `numpy.polynomial.polynomial.polyfit`, if I'm reading that manual page correctly. The docstring is non-specific as to how the weighting should occur. I'd like to change to your configuration, but worry about breaking people's code. Maybe it would be best to document this difference?

Reply:
Yes, this is an unfortunate situation. I agree that just changing the definition may lead to breaking other people's code in hard-to-track-down ways. So proper documentation seems best for now.

However, this is also numerically worse. Let `W = V' * V`, then the condition of `vand' * W * vand` is squared compared to `V * vand`. So even if this changes solving the system to using QR, it's still solving an unnecessarily badly conditioned system. Maybe we could change the `W::AbstractVector` method (which I assume is by far the most common case) to
```julia
_wlstsq(vand, y, W::AbstractVector) = qr(Diagonal(sqrt.(W)) * vand) \ (Diagonal(sqrt.(W)) * y)
```
That should be equivalent to the old code, but numerically better behaved. Of course, it fails for negative values in `W`, but that feels like a misuse of `W` anyway and it wouldn't silently lead to different results, but instead loudly throw.

Reply:
Yes, this slipped in during the big 1.0 overhaul, and will have to exit gracefully at 2.0. Until then, I'll add your suggestion and make a note in the documentation. Thanks again!!